src/modules/kvs-watch/kvs-watch.c

```diff
@@ -229,6 +371,42 @@ static void watcher_respond (struct namespace *ns, struct watcher *w)
         }
         w->rootseq = ns->commit->rootseq;
     }
+    /* flux_kvs_lookup (FLUX_KVS_WATCH)
+     *
+     * Ordering note: KVS lookups can be returned out of order, so lookup
+     * futures are added to the w->lookups zlist in commit order, and an
+     * out of order result is queued until all the lookups ahead of it have
+     * been retired.
+     *
```

Comment:
This comment sort of confused me.  I think it's b/c the latter half of what this comment says is implemented in the ```lookup_continuation``` function.  Perhaps it should be split up a bit?